client/game.cpp

```diff
@@ -190,4 +198,33 @@ void Game::Update(double delta, double mouse_x, double mouse_y, int window_width
 				+ tfm.other_time;
 	// clang-format on
 }
+
+void Game::ProcessEvents() {
+	EventQueue<KeyboardEvent>::ProcessEventQueue();
+	EventQueue<MouseClickEvent>::ProcessEventQueue();
+}
+
+void Game::On(std::shared_ptr<KeyboardEvent> data) {
+	if (data->action == KeyboardEvent::KEY_UP) {
+		switch (data->key) {
+		case GLFW_KEY_1: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[0])); break;
+		case GLFW_KEY_2: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[1])); break;
+		case GLFW_KEY_3: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[2])); break;
+		case GLFW_KEY_4: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[3])); break;
+		case GLFW_KEY_5: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[4])); break;
+		case GLFW_KEY_6: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[5])); break;
+		case GLFW_KEY_7: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[6])); break;
+		case GLFW_KEY_8: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[7])); break;
+		case GLFW_KEY_9: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[8])); break;
+		case GLFW_KEY_0: this->placement.SetMesh(MeshMap::Get(this->placeable_meshes[9])); break;
+		default: break;
+		}
+	}
+}
```

Comment:
This looks okay for this PR, but it's precisely the kind of thing that Lua should be capable of handling.

Reply:
Good call.

Reply:
I will tackle this in a separate PR as there is some work to wire keyboard events to lua and also it would be nice for the server to pass this down as a event.